azure-templates/stages.yml

```diff
@@ -48,6 +52,53 @@ parameters:
 
 # Note: If custom build steps are needed for a particular custom device, copy this entire template into the azure-pipeline and add steps before/after as needed
 stages:
+  - stage: DiffVIs${{ parameters.stageName }}
+    dependsOn: []
+    condition: and(eq(variables['Build.Reason'], 'PullRequest'), eq('${{ parameters.disableDiff }}', false))
+    pool:
+      name: DevCentral-VeriStand-CustomDevices
+      demands:
+        - agent.os -equals Windows_NT
+    jobs:
+      - job: Find64bitVersion
```

Comment:
This job is a bit redundant, but it is the best way I found to keep the compile-time variables in the `steps-pre-build.yml` template from rendering correctly in the step names when the pipeline runs.

![image](https://user-images.githubusercontent.com/31290917/230202360-3fe803d1-2fa4-4a57-83c8-a98c28987d6c.png)

Alternatively, the compile-time display names of steps in pre-build (like [variable assignment](https://github.com/ni/niveristand-custom-device-build-tools/blob/d558c8729a92f62010d65accc20f038a53f2da9c/azure-templates/steps-pre-build.yml#L20) and [cache deletion](https://github.com/ni/niveristand-custom-device-build-tools/blob/d558c8729a92f62010d65accc20f038a53f2da9c/azure-templates/steps-pre-build.yml#L125)) could be renamed and use run-time variables with prints in the steps for debug outputs. Then the diff stage could condense to one job:

```
    jobs:
      - job: DiffVIsforPR
        displayName: Diff VIs for PR
        variables:
          - group: niveristand-custom-devices
        steps:
          - powershell: |
              $lvVersionsToBuild = '${{ convertToJson(parameters.lvVersionsToBuild) }}' | ConvertFrom-Json
              ForEach ($lvVersionToBuild in $lvVersionsToBuild)
              {
                if ($lvVersionToBuild.bitness -eq "64bit")
                {
                  $diffBitness = $lvVersionToBuild.bitness
                  $diffVersion = $lvVersionToBuild.version
                  Write-Host "##vso[task.setvariable variable=lvDiffBitness]$diffBitness"
                  Write-Host "##vso[task.setvariable variable=lvDiffVersion]$diffVersion"
                }
              }
            name: FindLast64bitVersion
          - template: steps-pre-build.yml # Configure variables, check out repos, Clear cache
            parameters:
              lvVersionToBuild:
                version:               $(lvDiffVersion)
                bitness:               $(lvDiffBitness)
              releaseVersion:          ${{ parameters.releaseVersion }}
              archiveLocation:         ${{ parameters.archiveLocation }}
              buildOutputLocation:     ${{ parameters.buildOutputLocation }}
              codegenVis:              ${{ parameters.codegenVis }}
          - template: steps-diff.yml
            parameters:
              lvVersionToDiff:         $(lvDiffVersion)
```

Reply:
Converting the variable definition to something that encompasses multiple stages would also be useful for slimming down the custom packaging performed in SEECD, which redefines a handful of variables.  We could also split the pre-build-steps into 2 templates, one that is needed for diff/seecd packaging, and the other that only needs to run once.

Reply:
After talking with @papowerNI decided to defer large-scale rework of the stages and template files for later tech debt cleanup.